src/checkers/inference/solver/backend/AbstractFormatTranslator.java

```diff
@@ -24,15 +24,18 @@
 import checkers.inference.util.ConstraintVerifier;
 
 /**
- * Created by mier on 08/11/17.
+ * Abstract implementation of FormatTranslator that all concrete FormatTranslator should extend from.
+ * It has the necessary lattice that will be used and already setup the abstract structure and provided
```

Comment:
Better to clarify what "abstract structure" it setup.

Also, maybe make it explicit here that if a subclass doesn't override a serialize method here, by default it assumes this sub-class does not support corresponding constraint/slot.

Reply:
"abstract structure" really means the template of what a FormatTranslator looks like. But it looks too cumbersome for me to say, setup the seven encoder fields, factory methods that instantiate them and provides defaultImplementation etc.
I think the second point is reflected in the following documentation, isn't it?